### ISSUE TITLE
Is is possible to create nested/subcommands?

### ISSUE THREAD

**Issue by cdbkr** [jmcdo29/nest-commander#44]
I am trying to achieve nested/subcommands but it's a bit challenging.
Is it possible at the moment, or it would need to be implemented?

e.g.
`my-cli pizza create order --verbose`

commander is supporting it through the following approach:
https://github.com/tj/commander.js/blob/master/examples/nestedCommands.js

**Comment by jmcdo29** [comment]
This is something that would need to be implemented. Currently `nest-commander` really expects to not have nested commands, but I could see the use case for them (i.e. `docker compose up`). I'll need to think on how to architect something like this. If you've got an idea though I'd love to hear it

**Comment by jmcdo29** [comment]
Okay, what are the thoughts on something like 
```ts
@Command('up', {  parent: 'compose' })
export class UpCommand implements CommandRunner {}
```
For the command `docker compose up` (where `docker` is the `bin` name). Then under the hood, the `CommandRunnerService` will figure out if this is a sub command via the `parent` option. Everything else would still work as it already does.

What I need to know before going on implementing something like this, can the parent command have its own implementation, or is it more for functional separation?

**Comment by cdbkr** [comment]
thanks for the clarification! 
Probably following the Module approach embraced by NestJS wouldn't play well with Commander's subcommand features, because it's like you stated in the last comment: even parent's command can have its own implementation.
The Example you pasted looks good to me.
Instead, what do you think about the opposite, leaving the parents to register the first level subcommands?
```
@Command('up')
export class UpCommand implements CommandRunner {}

@Command('compose', {  subcommands: [ UpCommand ]})
export class ComposeCommand implements CommandRunner {}
```

**Comment by jmcdo29** [comment]
I actually think I like the opposite better, and it would lead to better self-documented code so that all of the child commands are referenced via the parent. It's gonna take a bit of work, but I think I can make something happen with this.

**Comment by jmcdo29** [comment]
I'm getting around to implementing this and had a thought: what if we also introduced a new decorator, `@SubCommand()` that takes in the same args as `@Command()`, but has different metadata so that it doesn't possibly get registered as both a sub command and a regular command? This would also help developers look quickly and see that the command being worked on is the child command of another.

**Comment by jmcdo29** [comment]
closed by #67. Available in 2.3.0
